singupy/singukafka.py

```diff
@@ -0,0 +1,404 @@
+        self.bootstrap_servers = os.environ.get('KAFKA_HOST', "my-cluster-kafka-bootstrap.kafka")
+
+    # Method: Initializes Kafka comsumer.
+    def init_consumer(self):
+        try:
+            self.consumer = KafkaConsumer(bootstrap_servers=self.bootstrap_servers,
+                                          group_id=self.group_id,
+                                          value_deserializer=lambda x: loads(x.decode('utf-8')),
+                                          auto_offset_reset=self.auto_offset_reset,
+                                          enable_auto_commit=self.enable_auto_commit)
+            log.info(" Kafka consumer connection established.")
+        except Exception as e:
+            log.exception(f"Error: Kafka Consumer connection failed with message: '{e}'.")
+            sys.exit(1)
+
+    # Method: Initializes Kafka producer.
+    def init_producer(self):
+        try:
+            self.producer = KafkaProducer(bootstrap_servers=self.bootstrap_servers,
+                                          value_serializer=lambda x: dumps(x).encode('utf-8'))
+            log.info("Kafka producer connection established.")
+        except Exception as e:
+            log.exception(f"Kafka producer connection failed with message: '{e}'.")
+            sys.exit(1)
+
+    # Method: Dummy Poll, which is needed to force assignment of partitions after subsribtion to topics.
+    def init_topic_partitions(self):
+        # TODO check if using a dummy poll is the correct way
+        # TODO handle that when making dummy poll, if auto comit is enabled, then the offset will shift unessecary?
+        try:
+            self.consumer.poll(timeout_ms=self.poll_timeout_ms)
+            log.info("Initial poll done. TopicPartitions are now assigned.")
+        except Exception as e:
+            log.exception(f"Initial poll failed with message '{e}'.")
+            sys.exit(1)
+
+    # Method: Subscribes consumer to topics
+    def subscribe_topics(self):
+        try:
+            self.consumer.subscribe(self.topics_consumed_list)
+            log.info(f"Kafka consumer subscribed to topics: '{self.topics_consumed_list}'.")
+        except Exception as e:
+            log.exception(f"Kafka consumer subscribtion to topics: '{self.topics_consumed_list}' " +
+                          f"failed with message: '{e}'.")
+            sys.exit(1)
+
+    # Method: Verify if topics exist in Kafka Brooker.
+    def verify_topic_existence(self):
+        unavbl_topics = []
+        for topic in self.topic_list:
+            try:
+                if topic not in self.consumer.topics():
+                    unavbl_topics.append(topic)
+            except Exception as e:
+                log.exception(f"Verifying if topic: '{topic}' exists failed with message: '{e}'.")
+                sys.exit(1)
+
+        if unavbl_topics:
+            log.exception(f"The Topic(s): {unavbl_topics} does not exist.")
+            sys.exit(1)
```

Comment:
```suggestion
    def verify_topic_existence(self) -> bool:
        unavbl_topics = []
        for topic in self.topic_list:
            try:
                if topic not in self.consumer.topics():
                    unavbl_topics.append(topic)
            except Exception as e:
                log.exception(f"Verifying if topic: '{topic}' exists failed with message: '{e}'.")
                sys.exit(1)

        if unavbl_topics:
            log.warning(f"The topics {unavbl_topics} do not exist.")
            return False
        else:
            return True
```
Are we sure this should exit? What if it is a producer-topic that has not been written to yet and it will be created when we start writing? Perhaps it should just return false and then let the calling action decide what to do about it?